tests/css/test_styles.py

```diff
@@ -185,3 +194,77 @@ def test_widget_style_size_fails_if_data_type_is_not_supported(size_dimension_in
 
     with pytest.raises(StyleValueError):
         widget.styles.width = size_dimension_input
+
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize(
+    "overflow_y,scrollbar_gutter,text_length,expected_text_widget_width,expects_vertical_scrollbar",
+    (
+        # ------------------------------------------------
+        # ----- Let's start with `overflow-y: auto`:
```

Comment:
hopefully I've described these test cases well enough to make this test maintainable :-) :crossed_fingers: